lib/friendly_shipping/services/ship_engine/serialize_address_validation_request.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module FriendlyShipping
+  module Services
+    class ShipEngine
+      class SerializeAddressValidationRequest
+        # @param locations [Array<Physical::Location>]
+        # @return [Hash]
+        def self.call(locations:)
+          locations.map do |location|
+            {
+              name: location.name,
+              phone: location.phone,
+              email: location.email,
+              company_name: location.company_name,
+              address_line1: location.address1,
+              address_line2: location.address2,
+              address_line3: location.address3,
+              city_locality: location.city,
+              state_province: location.region.code,
+              postal_code: location.zip,
+              country_code: location.country.code,
+              address_residential_indicator: location.address_type == "residential" ? "yes" : "no"
+            }
```

Comment:
It doesn't error out if any of the attributes are nil like UPS does?? :)

Reply:
Nope. Shocking eh?